pandas/tests/test_resample.py

```diff
@@ -626,12 +626,7 @@ def test_asfreq(self, series_and_frame, freq):
         obj = series_and_frame
 
         result = obj.resample(freq).asfreq()
-        if freq == '2D':
-            new_index = obj.index.take(np.arange(0, len(obj.index), 2))
-            new_index.freq = to_offset('2D')
-        else:
-            new_index = self.create_index(obj.index[0], obj.index[-1],
-                                          freq=freq)
+        new_index = self.create_index(obj.index[0], obj.index[-1], freq=freq)
```

Comment:
@mroeschke 

The case noted [here](https://github.com/pandas-dev/pandas/pull/22488#issuecomment-415560778) is covered by this test. However, it used to produce the incorrect index so tests were passing (which is incorrect). Here is the old expected index for TDI:

```
In [1]: obj.index.take(np.arange(0, len(obj.index), 2))
Out[1]: TimedeltaIndex(['1 days', '3 days', '5 days', '7 days', '9 days'], dtype='timedelta64[ns]', freq='2D')
```